python/ray/data/tests/test_stats.py

```diff
@@ -1344,6 +1349,62 @@ def test_stats_actor_datasets(ray_start_cluster):
         assert value["state"] == "FINISHED"
 
 
+@patch.object(ray.data._internal.stats, "STATS_ACTOR_UPDATE_INTERVAL_SECONDS", new=0.5)
+@patch.object(StatsManager, "_stats_actor_handle")
+@patch.object(StatsManager, "UPDATE_THREAD_INACTIVITY_LIMIT", new=1)
+def test_stats_manager(shutdown_only):
```

Comment:
nice test. one minor concern is that it looks too stressful for a unit test. If you find it slow, consider making the workload simpler. For example, dataset can be smaller, like `ray.data.range(10, parallelism=10).map_batches(...).iter_rows()`